cmake/DetectOptions.cmake

```diff
@@ -606,6 +606,21 @@ elseif(ADIOS2_USE_Campaign)
   endif()
 endif()  
 
+# KVCache
+if(ADIOS2_USE_Cache STREQUAL AUTO)
+    find_package(hiredis REQUIRED)
```

Comment:
I think this use of find_package() should not have the REQUIRED keyword.

Reply:
I second this, when ADIOS2_USE_Cache is AUTO we should not use required. REQUIRED is only with =ON

Reply:
ok I have removed it and finished formatting by docker

Reply:
Ah, formatting is OK except for shell scripts.  It's picky about shell scripts and as far as I know those have to be fixed manually.  Unless you know an automatic way @vicentebolea ?  @Change72 click on "details" next to the format line below to see what it is complaining about.

Reply:
@eisenhauer is correct, CI uses shellcheck for ensuring best practices for shell scripts, we do not force any formatting/indentation style for shell scripts.